File: cpp/44_football.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin >> s;
    int cnt = 1;
    for (size_t i = 1; i < s.size(); i++)
    {
        if (s[i] == s[i - 1])
        {
            cnt++;
            if (cnt >= 7)
            {
                cout << "YES\n";
                return 0;
            }
        }
        else
            cnt = 1;
    }
    cout << "NO\n";
}

File: python/44_football.py
s = input().strip()
cnt = 1
for i in range(1, len(s)):
    if s[i] == s[i - 1]:
        cnt += 1
        if cnt >= 7:
            print("YES")
            break
    else:
        cnt = 1
else:
    print("NO")
